mala/interfaces/ase_calculator_uncertainty.py

```diff
@@ -0,0 +1,157 @@
+
+        Parameters
+        ----------
+        run_name : str
+            Name under which the model is saved.
+
+        path : str
+            Path where the model is saved.
+        """
+        loaded_params, loaded_network, \
+            new_datahandler, loaded_runner = MultiPredictor.\
+            load_run(run_name, path=path)
+        calculator = cls(loaded_params, loaded_network, new_datahandler,
+                         predictor=loaded_runner)
+        return calculator
+
+    def calculate(self, atoms=None, properties=['energy'],
+                  system_changes=all_changes):
+        """
+        Perform the calculations.
+
+        Parameters
+        ----------
+        atoms : ase.Atoms
+            Atoms object for which to perform the calculation.
+
+        properties: list of str
+            List of what needs to be calculated.  Can be any combination
+            of 'energy', 'forces', 'stress', 'dipole', 'charges', 'magmom'
+            and 'magmoms'.
+
+        system_changes: list of str
+            List of what has changed since last calculation.  Can be
+            any combination of these six: 'positions', 'numbers', 'cell',
+            'pbc', 'initial_charges' and 'initial_magmoms'.
+        """
+        barrier()
+        Calculator.calculate(self, atoms, properties, system_changes)
+
+        # Get the LDOS from the NN.
+        ldos = self.predictor.predict_for_atoms(atoms)
+        energies = []
+
+        for m in range(0, self.predictor.number_of_models):
+            # forces = np.zeros([len(atoms), 3], dtype=np.float64)
+
+            # If an MPI environment is detected, ASE will use it for writing.
+            # Therefore we have to do this before forking.
+            self.data_handler.\
+                target_calculator.\
+                write_tem_input_file(atoms,
+                                     self.data_handler.
+                                     target_calculator.qe_input_data,
+                                     self.data_handler.
+                                     target_calculator.qe_pseudopotentials,
+                                     self.data_handler.
+                                     target_calculator.grid_dimensions,
+                                     self.data_handler.
+                                     target_calculator.kpoints)
+
+            ldos_calculator: LDOS = self.data_handler.target_calculator
+
+            ldos_calculator.read_from_array(ldos[m])
+            energy, self.last_energy_contributions \
+                = ldos_calculator.get_total_energy(return_energy_contributions=
+                                                   True)
+            energies.append(energy)
+            barrier()
+
+
+        # Use the LDOS determined DOS and density to get energy and forces.
+        self.results["energy"] = np.mean(energies)
+        self.results["energy_uncertainty"] = np.std(energies)
+        # if "forces" in properties:
```

Comment:
It would be useful to keep `energies` for external access, e.g. `results["energy_samples"]`.

Reply:
Good point. I implemented this.